### PR TITLE
Added Include note about class being obsolete

### PR DESCRIPTION
**Pull request by lorieide**
## Summary

Describe your changes here.

Fixes #Issue_Number (if available)

### CODE REVIEW

**Comment by gewarren** [comment]
@lorieide What's the motivation behind this PR?

**Comment by lorieide** [comment]
> @lorieide What's the motivation behind this PR?

@gewarren Sure. I've been assigned this GitHub issue to resolve https://github.com/MicrosoftDocs/azure-docs/issues/111552

**Comment by lorieide** [comment]
@gewarren Here's what I found:
The .NET WebRequest and HttpWebRequest classes are indeed considered obsolete and are recommended to be replaced with HttpClient. This is in line with modern development practices and the evolution of the .NET framework. Specifically, as per the Microsoft documentation, starting from .NET 5+, it is advised to use HttpClient instead of WebRequest, HttpWebRequest, ServicePoint, and WebClient.
I validated this information is correct as all of these topics have a scattered Important note about this development.
I recommend adding a warning note as an include below to all of the topics in the WebRequest, ServicePoint, WebClient, and HttpWebRequest Class topics.



Repro Steps



System Info



Discussion


Add a comment. Use # to link a work item, @ to mention a person, or ! to link a pull request.
[Markdown supported.](http://go.microsoft.com/fwlink/?LinkId=823918)
Paste or select files to insert.
switch to HTML editor
Lori Eide
[commented 2h ago](https://dev.azure.com/msft-skilling/Content/_queries/edit/205230/?queryId=296f6524-f728-4969-813b-d09bbf90097c#4501565)




@Barclay Neira I've updated the each of the class topics in the 4 XML files with an include mentioned below. Waiting for response from the workflow. I've not worked in the .NET repo before, so not sure how this process goes.

https://github.com/dotnet/dotnet-api-docs/pull/10015 

Lori Eide
[commented 4h ago (edited)](https://dev.azure.com/msft-skilling/Content/_queries/edit/205230/?queryId=296f6524-f728-4969-813b-d09bbf90097c#4501468)




Found this on Copilot:

The .NET WebRequest and HttpWebRequest classes are indeed considered obsolete and are recommended to be replaced with HttpClient. This is in line with modern development practices and the evolution of the .NET framework. Specifically, as per the Microsoft documentation, starting from .NET 5+, it is advised to use HttpClient instead of WebRequest, HttpWebRequest, ServicePoint, and WebClient.

I validated this information is correct as all of these topics have a scattered Important note about this development.

I recommend adding the note as an include below to all of the topics in the WebRequest, ServicePoint, WebClient, and HttpWebRequest Class topic

**Comment by gewarren** [comment]
@lorieide Ahh, okay. Currently, if a .NET API is marked with the ObsoleteAttribute, for example, [WebRequest.Create](https://github.com/dotnet/runtime/blob/a08fe0fc6cc42cf0764a0e25e31c9d464d060d86/src/libraries/System.Net.Requests/src/System/Net/WebRequest.cs#L149-L155), then the API reference docs [automatically show the CAUTION block](https://learn.microsoft.com/en-us/dotnet/api/system.net.webrequest.create?view=net-8.0#system-net-webrequest-create(system-string)) with the text from the ObsoleteAttribute. So that's why you see the info on some APIs and not others. There's also a [work item to propagate the obsolete warning to all members of a type](https://dev.azure.com/ceapex/Engineering/_workitems/edit/903118) that's marked obsolete (although I don't think that applies to any of these 4 System.Net types unfortunately).

Related to https://github.com/dotnet/dotnet-api-docs/issues/6867.

File: xml/System.Net/WebRequest.xml [hunk 1]
```diff
@@ -1324,6 +1365,9 @@ Note: In <see href="https://go.microsoft.com/fwlink/?LinkID=247912">.NET for Win
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
+
+[!INCLUDE [obsolete-notice](~/includes/web-request-obsolete.md)]
```
Comment (gewarren) [inline xml/System.Net/WebRequest.xml:1369]: Not needed here. This API already has an obsolete banner.

File: xml/System.Net/WebRequest.xml [hunk 2]
```diff
@@ -1771,6 +1830,9 @@ This property allows an application to determine which <xref:System.Net.IWebRequ
...
```
Comment (gewarren) [inline xml/System.Net/WebRequest.xml:1834]: Not needed here. This API already has an obsolete banner.

**Review by gewarren — APPROVED** [review]

**Comment by MihaZupan** [comment]
FYI `ServicePointManager` was just marked as obsolete in https://github.com/dotnet/runtime/pull/103456.
And while types like `ServicePoint`/`WebRequest` aren't explicitly marked as obsolete, they effectively are since all the ways to obtain their instances (`ServicePointManager.FindServicePoint`, `WebRequest.Create`) are obsolete.

File: xml/System.Net/HttpWebRequest.xml [hunk 3]
```diff
@@ -2998,6 +3087,9 @@ The GetHashCode method returns a hash code of the web request. This value can be
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
+
+[!INCLUDE [obsolete-notice](~/includes/http-web-request-obsolete.md)]
```
Comment (gewarren) [inline xml/System.Net/HttpWebRequest.xml]:
```suggestion
```

File: xml/System.Net/HttpWebRequest.xml [hunk 4]
```diff
@@ -4865,6 +5020,9 @@ The GetHashCode method returns a hash code of the web request. This value can be
...
```
Comment (gewarren) [inline xml/System.Net/HttpWebRequest.xml]:
```suggestion
```

File: xml/System.Net/WebClient.xml [hunk 5]
```diff
@@ -176,9 +175,10 @@
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
- The parameterless constructor creates a new instance of the <xref:System.Net.WebClient> class. The default HTTP method is GET. The default FTP method is RETR. The default <xref:System.Net.WebClient.Encoding%2A> is <xref:System.Text.Encoding.Default%2A>. The default value of <xref:System.Net.HttpWebRequest.AllowAutoRedirect%2A> is `true`.
 
+[!INCLUDE [obsolete-notice](~/includes/web-client-obsolete.md)]
```
Comment (gewarren) [inline xml/System.Net/WebClient.xml]:
```suggestion
```

File: xml/System.Net/WebClient.xml [hunk 6]
```diff
@@ -242,6 +242,9 @@
           <format type="text/markdown"><![CDATA[
 
 ## Remarks
+
+[!INCLUDE [obsolete-notice](~/includes/web-client-obsolete.md)]
```
Comment (gewarren) [inline xml/System.Net/WebClient.xml]:
```suggestion
```

File: xml/System.Net/WebClient.xml [hunk 7]
```diff
@@ -299,6 +302,9 @@
...
```
Comment (gewarren) [inline xml/System.Net/WebClient.xml]:
```suggestion
```

File: xml/System.Net/WebRequest.xml [hunk 8]
```diff
@@ -1024,6 +1053,9 @@ Note: In <see href="https://go.microsoft.com/fwlink/?LinkID=247912">.NET for Win
...
```
Comment (gewarren) [inline xml/System.Net/WebRequest.xml]:
```suggestion
```

File: xml/System.Net/WebRequest.xml [hunk 9]
```diff
@@ -1118,6 +1150,9 @@ Note: In <see href="https://go.microsoft.com/fwlink/?LinkID=247912">.NET for Win
...
```
Comment (gewarren) [inline xml/System.Net/WebRequest.xml]:
```suggestion
```

File: xml/System.Net/WebRequest.xml [hunk 10]
```diff
@@ -1196,6 +1231,9 @@ Note: In <see href="https://go.microsoft.com/fwlink/?LinkID=247912">.NET for Win
...
```
Comment (gewarren) [inline xml/System.Net/WebRequest.xml]:
```suggestion
```

File: xml/System.Net/WebRequest.xml [hunk 11]
```diff
@@ -1266,6 +1304,9 @@ Note: In <see href="https://go.microsoft.com/fwlink/?LinkID=247912">.NET for Win
...
```
Comment (gewarren) [inline xml/System.Net/WebRequest.xml]:
```suggestion
```

File: xml/System.Net/WebRequest.xml (hunk 1 above)
Comment (gewarren) [inline xml/System.Net/WebRequest.xml]:
```suggestion
```

**Pull request merged**